app/services/registration_transfer_service.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class RegistrationTransferService
+  def initialize(registration)
+    @registration = registration
+    @transient_registration = find_transient_registration
+  end
+
+  def transfer_to_user(email)
+    @recipient_user = find_user(email)
+    return :no_matching_user if @recipient_user.blank?
+
+    update_account_email_for(@registration)
+    update_account_email_for(@transient_registration) if @transient_registration.present?
+
+    send_confirmation_email
+
+    :success_existing_user
+  end
+
+  private
+
+  def find_transient_registration
+    WasteCarriersEngine::TransientRegistration.where(reg_identifier: @registration.reg_identifier).first
+  end
+
+  def find_user(email)
+    return nil unless email.present?
+
+    ExternalUser.where(email: email).first
+  end
+
+  def update_account_email_for(registration)
+    registration.account_email = @recipient_user.email
+    registration.save!
+  end
+
+  def send_confirmation_email
+    RegistrationTransferMailer.transfer_to_existing_account_email(@registration).deliver_now
```

Comment:
Should have said this in previous PR's, but can we wrap this in a `begin/rescue`. Sending emails where we're not offloading them to an ActiveJob can sometimes be flaky (based on experience with other services). At this point the transfer would have been successful but the NCCC user would not see that if sending the email failed. They would just see an error message.

Catching the exception and logging it in Errbit means *we* still know something went wrong, but the main thing is the user gets to see the confirmation that the transfer is complete so no confusion.

Reply:
Good point, will sort this out (and make a note to fix it in our other mailers too).

Reply:
Issue for the other mailers in the engine: https://github.com/DEFRA/waste-carriers-renewals/issues/293